backend/substrapp/compute_tasks/image_builder.py

```diff
@@ -129,17 +131,29 @@ def _build_container_image(path: str, tag: str) -> None:  # noqa: C901
                 f"Error creating pod {NAMESPACE}/{pod_name}. Reason: {e.reason}, status: {e.status}, body: {e.body}"
             ) from e
 
+    build_exc = None
+
     try:
         watch_pod(k8s_client, pod_name)
+
     except Exception as e:
-        # In case of concurrent build, it may fail
-        # check if image exists
-        if not container_image_exists(tag):
-            raise compute_task_errors.BuildError from e
+        # In case of concurrent builds, it may fail. Check if the image exists.
+        if container_image_exists(tag):
+            return
```

Comment:
slightly out of the scope of this PR but it would be nice readability to change the negation in the value evaluation [`l. 121`](https://github.com/Substra/substra-backend/blob/main/backend/substrapp/compute_tasks/image_builder.py#L121)
````
    create_pod = not pod_exists(k8s_client, pod_name)
    if create_pod:
           <<create the pod>>
````

to something like:

````
    pod_exists = pod_exists(k8s_client, pod_name)
    if ! pod_exists:
           <<create the pod>>
````

Reply:
but then here you would have a `not pod_exists` and that would not make much sense: https://github.com/Substra/substra-backend/blob/23184cd721d3f8de25f840ce6976074ddad17fbf/backend/substrapp/compute_tasks/image_builder.py#L140

Reply:
IMO it makes it simpler to have a variable that describes "what needs to be done" (create_pod) rather than "why the thing needs to be done" (not pod_exists)